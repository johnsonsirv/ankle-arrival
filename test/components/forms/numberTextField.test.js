/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import NumberTextField from '../../../src/components/forms/inputs/numberTextField';

function setup() {
  const props = {
    name: 'number',
    onChange: jest.fn(),
  };
  const enzymeWrapper = mount(<NumberTextField {...props} />);

  return { props, enzymeWrapper };
}

describe('<NumberTextField /> rendering', () => {
  const { enzymeWrapper } = setup();
  it('renders correctly', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should render 1 editable <input> number', () => {
    expect(enzymeWrapper.find('input[type="number"]')).toHaveLength(1);
  });
});

describe('<NumberTextField /> interactions', () => {
  const { props, enzymeWrapper } = setup();
  it('should handle onChange', () => {
    enzymeWrapper.simulate('change', {
      target: { value: 123 },
    });
    expect(props.onChange).toHaveBeenCalled();
  });
});
