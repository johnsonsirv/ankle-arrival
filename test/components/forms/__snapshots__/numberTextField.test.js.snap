// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberTextField /> rendering renders correctly 1`] = `ReactWrapper {}`;
